pkg/actors/reminders/reminders.go

```diff
@@ -159,9 +160,37 @@ func (r *reminders) CreateReminder(ctx context.Context, reminder *internal.Remin
 
 	stop := make(chan struct{})
 
-	err = r.storeReminder(ctx, store, reminder, stop)
+	config := retry.DefaultConfig()
+	config.Multiplier = 1.0
+	b := config.NewBackOffWithContext(ctx)
+
+	var etagErr *state.ETagError
+	err = retry.NotifyRecover(func() error {
+		innerErr := r.storeReminder(ctx, store, reminder, stop)
+		if innerErr != nil {
+			if errors.As(innerErr, &etagErr) {
+				switch etagErr.Kind() {
+				case state.ETagMismatch:
+					// If the etag is mismatched, we can retry the operation.
+					return innerErr
+				default:
+					log.Errorf("error storing reminder: %w", innerErr)
+					err = innerErr
```

Comment:
Please don't do this: it's a side effect that will cause race conditions.

Instead, please use `NotifyRecoverWIthData` to return a second error if needed.

Reply:
Actually even better... wrap the non-etag error in a permanent error: https://pkg.go.dev/github.com/cenkalti/backoff/v4#PermanentError we do that in other parts of Dapr.

Reply:
done